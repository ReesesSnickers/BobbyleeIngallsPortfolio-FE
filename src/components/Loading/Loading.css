.portfolio-loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  text-align: center;
  transition: margin 0.3s linear;
}
h1 {
  font-size: 40px;
  font-family: Helvetica, Arial, sans-serif;
  color: #0e1a3c;
  margin: 0px 40px;
  transition: font 0.3s linear, margin 0.3s linear;
}

@media screen and (max-width: 930px) {
  .portfolio-loading {
    margin-top: 200px;
    transition: margin 0.3s linear;
  }
}

@media screen and (max-width: 570px) {
  h1 {
    font-size: 30px;
    margin: 0px 40px;
    transition: font 0.3s linear, margin 0.3s linear;
  }
}

@media screen and (max-width: 430px) {
  h1 {
    font-size: 25px;
    margin: 0px 40px;
    transition: font 0.3s linear, margin 0.3s linear;
  }
}
