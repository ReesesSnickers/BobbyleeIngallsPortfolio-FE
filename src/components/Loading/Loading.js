import React from "react";
import PropTypes from "prop-types";

const Loading = ({ load }) => {
  return (
    <section className="portfolio-loading">
      <h1>Sorry. Please standby while the {load} are retrieved.</h1>
    </section>
  );
};

export default Loading;

Loading.propTypes = {
  // text of what is being retrieved
  load: PropTypes.string.isRequired
};
