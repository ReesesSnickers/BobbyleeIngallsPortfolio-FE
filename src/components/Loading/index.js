import "./Loading.css";
export { default } from "./Loading";
