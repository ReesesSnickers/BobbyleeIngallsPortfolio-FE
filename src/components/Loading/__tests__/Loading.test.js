import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Loading from "../Loading.js";

let props;
let mountedComponent;

describe("Loading", () => {
  const testComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<Loading {...props} />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    props = {
      load: "MockLoad"
    };
    mountedComponent = undefined;
  });

  // initial render
  it("Renders the Loading component", () => {
    let button = testComponent().find("section.portfolio-loading");
    expect(button.length).toBe(1);
  });

  // snapshot
  it("should match snapshot", () => {
    const tree = renderer.create(<Loading {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // load
  it("When load is defined", () => {
    const h1 = testComponent().find("h1");
    expect(h1.text()).toBe(
      "Sorry. Please standby while the MockLoad are retrieved."
    );
  });
});
