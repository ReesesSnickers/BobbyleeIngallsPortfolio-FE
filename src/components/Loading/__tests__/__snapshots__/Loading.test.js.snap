// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match snapshot 1`] = `
<section
  className="portfolio-loading"
>
  <h1>
    Sorry. Please standby while the 
    MockLoad
     are retrieved.
  </h1>
</section>
`;
